webapp/graphite/functions/params.py

```diff
@@ -7,54 +7,83 @@
 
 
 class ParamTypes(object):
-  pass
+    pass
 
 
 class ParamType(object):
-  options = []
+    options = []
 
-  def __init__(self, name, validator=None):
-    self.name = name
-    self.validator = validator
+    def __init__(self, name, validator=None):
+        self.name = name
+        self.validator = validator
 
-  @classmethod
-  def register(cls, name, *args):
-    setattr(ParamTypes, name, cls(name, *args))
+    @classmethod
+    def register(cls, name, *args):
+        setattr(ParamTypes, name, cls(name, *args))
 
-  def isValid(self, value):
-    if self.validator is None:
-      # if there's no validator for the type we assume True
-      return True
+    def isValid(self, value):
+        if self.validator is None:
+            # if there's no validator for the type we assume True
+            return value
 
-    return self.validator(value)
+        return self.validator(value)
 
 
 def validateBoolean(value):
-  return isinstance(value, bool)
+    if isinstance(value, six.string_types):
+        if value.lower() == 'true':
+            return True
+        if value.lower() == 'false':
+            return False
+        raise ValueError('Invalid boolean value: {value}'.format(value=repr(value)))
+
+    if type(value) in [int, float]:
+        if value == 0:
+            return False
+        return True
```

Comment:
Maybe this should not accept all non-zero numeric values as `True`? maybe just `1` to go along with `0`?

I personally lean towards no numbers here, just `True`/`False` or `"true"`/`"false"`. While good error responses with found/expected types are definitely great, I fear that too much lenience and auto-conversion may ultimately be more confusing - things may seem to work but the results are totally wrong, or return an error that doesn't make sense, and it's because there was a mixup in a different argument which was auto-converted.

Reply:
Good point inre being too permissive, I think only 0/1 is a good compromise here.

Reply:
sounds good 04ec05f6b6c37dc520e59316c818890666b99e9f

Reply:
looks good, thanks :)